### PR TITLE
Mint challenge NFT once a challenge is resolved

### PR DESCRIPTION
**Pull request by ezynda3**

### CODE REVIEW

File: packages/buidler/contracts/facets/ChallengesFacet.sol [hunk 1]
```diff
@@ -386,6 +392,7 @@ contract ChallengesFacet is BaseRelayRecipient {
         } else {
             _challenge.status == Status.Refund;
         }
+        challengeToken().firstMint(_challenge.athlete, _challenge.challengeUrl, _challenge.jsonUrl);
```
Comment (youngkidwarrior) [inline packages/buidler/contracts/facets/ChallengesFacet.sol:395]: Nothing should get minted on Status.Refund. Refunds mean the Challenge was never completed

**Review by youngkidwarrior — CHANGES_REQUESTED** [review]

File: packages/buidler/contracts/ChallengeToken.sol [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@opengsn/gsn/contracts/BaseRelayRecipient.sol";
+import "./IChallengeRegistry.sol";
+import "./IChallenges.sol";
+import "./SignatureChecker.sol";
+
+contract ChallengeToken is BaseRelayRecipient, ERC721, SignatureChecker {
+
+    constructor() ERC721("Shenanigan Challenge", "SHE") public {
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply packages/buidler/contracts/ChallengeToken.sol:14]: So the token name should be "CHALLENGE"? Also the symbol?

File: packages/buidler/contracts/facets/ChallengesFacet.sol (hunk 1 above)
Comment (ezynda3) [reply packages/buidler/contracts/facets/ChallengesFacet.sol]: Fixed

File: packages/buidler/contracts/ChallengeToken.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (youngkidwarrior) [reply packages/buidler/contracts/ChallengeToken.sol:14]: the symbol is CHALLENGE, name is ok as is

[the remark this answers is not in the document]
Comment (ezynda3) [reply packages/buidler/contracts/ChallengeToken.sol]: Fixed

**Review by youngkidwarrior — APPROVED** [review]

**Pull request merged**